docs/understand/programming_model.rst

```diff
@@ -10,23 +10,25 @@ HIP programming model
 
 The HIP programming model makes it easy to map data-parallel C/C++ algorithms to
 massively parallel, wide single instruction, multiple data (SIMD) architectures,
-such as GPUs. A basic understanding of the underlying device architecture helps you
+such as GPUs. 
+
+While the model may be expressed in most imperative languages, (for example
+Python via PyHIP) this document will focus on the original C/C++ API of HIP.
+
+A basic understanding of the underlying device architecture helps you
 make efficient use of HIP and general purpose graphics processing unit (GPGPU)
 programming in general.
 
 RDNA & CDNA architecture summary
 ================================
 
-Most GPU architectures, like RDNA and CDNA, have a hierarchical structure.
-The innermost piece is a SIMD-enabled vector Arithmetic Logical Unit (ALU).
-In addition to the vector ALUs, most recent GPUs also house matrix ALUs for
-accelerating algorithms involving matrix multiply-accumulate operations.
-AMD GPUs also contain scalar ALUs, that can be used to reduce the load on the
-vector ALU by performing operations which are uniform for all threads of a warp.
+Most GPU architectures, like RDNA and CDNA, have a hierarchical structure (:ref:`rdna3_cu` and :ref:`cdna3_cu`). 
 
-A set of ALUs, together with register files, caches and shared memory, comprise
-a larger block, often referred to as a compute unit (CU), e.g. in OpenCL and
-AMD block diagrams, or as streaming multiprocessor (SM).
+A set of Arithmetic Logical Units (ALUs), together with register files, caches and
+shared memory, comprise a larger block, often referred to as a Compute Unit (CU)
+or as streaming multiprocessor. 
```

Comment:
Suggestion for fusing those remaining two paragraphs and providing an actual summary.

> GPUs in general are made up of basic building blocks called compute units (CUs), that execute the threads of a kernel. These CUs provide the necessary resources for the threads: the Arithmetic Logical Units (ALUs), register files, caches and shared memory for efficient communication between the threads. 
> 
> This design allows for efficient execution of kernels while also being able to scale from small GPUs embedded in APUs with few CUs up to GPUs designed for data centers with hundreds of CUs.
> Figure :numref:`rdna3_cu: and :numref:`cdna3_cu` show examples of such compute units.

Numref might not work here, as the pictures are not enumerated, but :ref: just pastes the figure caption there, which doesn't really look nice

Reply:
Added your suggestion.